src/tools/compiletest/src/main.rs

```diff
@@ -617,14 +617,28 @@ pub fn make_test(config: &Config, testpaths: &TestPaths) -> test::TestDescAndFn
         || (config.mode == DebugInfoGdb || config.mode == DebugInfoLldb)
             && config.target.contains("emscripten");
 
-    test::TestDescAndFn {
-        desc: test::TestDesc {
-            name: make_test_name(config, testpaths),
-            ignore,
-            should_panic,
-            allow_fail: false,
-        },
-        testfn: make_test_closure(config, testpaths),
+    if early_props.revisions.is_empty() {
+        vec![test::TestDescAndFn {
+            desc: test::TestDesc {
+                name: make_test_name(config, testpaths, None),
+                ignore,
+                should_panic,
+                allow_fail: false,
+            },
+            testfn: make_test_closure(config, testpaths),
+        }]
+    } else {
+        early_props.revisions.iter().map(|revision|
+            test::TestDescAndFn {
+                desc: test::TestDesc {
+                    name: make_test_name(config, testpaths, Some(revision)),
+                    ignore,
+                    should_panic,
+                    allow_fail: false,
+                },
+                testfn: make_test_closure(config, testpaths),
```

Comment:
I think we have to be threading the `revision` down into the test closure here...